### ISSUE TITLE
about L2 norm error

### ISSUE THREAD

**Issue by Zhenzh1219** [Matt2371/PINN_navier_stokes#2]
excuse me, why after I use your model2 to train, the result of calculating L2 norm using the test data mentioned in Raissi (2019) code is not very good?
data = scipy.io.loadmat('data/cylinder_wake.mat')
    U_star = data['U_star']  # N x dimension x T, dimension = 2
    P_star = data['p_star']  # N x T
    t_star = data['t']  # T x 1
    X_star = data['X_star']  # N x dimension
    # Test Data
    snap = np.array([100])
    TT = np.tile(t_star, (1, N)).T  # N x T,转置
    x_star = X_star[:, 0:1].reshape(-1, 1)
    y_star = X_star[:, 1:2].reshape(-1, 1)
    t_star = TT[:, snap].reshape(-1, 1)
    u_star = U_star[:, 0, snap].flatten().reshape(-1,1)
    v_star = U_star[:, 1, snap].flatten().reshape(-1,1)
    p_star = P_star[:, snap].flatten().reshape(-1,1)
    model_output = model(x_star, y_star, t_star)
    u_pred, v_pred, p_pred = model_output[0], model_output[1], model_output[2]
    u_pred, v_pred, p_pred = (u_pred.detach().cpu().numpy().flatten().reshape(-1, 1),
                              v_pred.detach().cpu().numpy().flatten().reshape(-1, 1),
                              p_pred.detach().cpu().numpy().flatten().reshape(-1, 1))
    error_u = np.linalg.norm(u_star - u_pred, 2) / np.linalg.norm(u_star, 2)
    error_v = np.linalg.norm(v_star - v_pred, 2) / np.linalg.norm(v_star, 2)
    error_p = np.linalg.norm(p_star - p_pred, 2) / np.linalg.norm(p_star, 2)
    
  And I need epochs=500, train_selection=0.05
  The result, does it perform well enough?:
  Error u: 9.317031e-03
  Error v: 2.138385e-02
  Error p: 2.404164e-02

**Comment by Matt2371** [comment]
Sorry, I cannot comment on using L2-norm error as you have defined. I provide a detailed analysis of root mean square error (rMSE) and the coefficient of determination (R2) using the Raissi (2019) data in my notebook file, evaluate_model.ipynb, please refer to the results there. Based on these metrics, and by comparing the predicted and reference flow fields, training 200 epochs with a train_selection of 0.005 is plenty and produces very accurate results throughout the entire flow field and simulation period. Thanks.

**Comment by Zhenzh1219** [comment]
Thank you very much, I will try some more
